File: DocProxy/docproxy/src/main/java/org/ls/docproxy/Investor.java
package org.ls.docproxy;

/**
 * Author by Less on 2019/6/9.
 */
public class Investor implements IInvestor {

  private String mName;

  public Investor(String name){
    this.mName = name;
  }

  @Override
  public void login(String user, String password) {
    System.out.println(this.mName + "登录成功！");
  }

  @Override
  public void buyStock() {
    System.out.println(this.mName + "在买股票！");
  }

  @Override
  public void sellStock() {
    System.out.println(this.mName + "在卖股票！");
  }
}

File: DocProxy/app/src/main/java/org/ls/docproxy/proxy/Investor.kt
package org.ls.docproxy.proxy

import android.util.Log

/**
 * 真正的投资者类
 */
class Investor(private var name: String) : IInvestor {

  override fun login(
    user: String,
    password: String
  ) {
    System.out.println("$name-登录成功")
  }

  override fun buyStock() {
    System.out.println("$name-在购买股票")
  }

  override fun sellStock() {
    System.out.println("$name-在卖股票")
  }

}